aries_cloudagent/messaging/jsonld/routes.py

```diff
@@ -45,31 +49,33 @@ async def sign(request: web.BaseRequest):
     response = {}
     try:
         context: AdminRequestContext = request["context"]
+        session = await context.session()
+        try:
+            session.inject(BaseWallet, required=True)
+        except Exception:
+            raise web.HTTPForbidden(reason="No wallet available")
```

Comment:
This is a little dirty and should be basically unreachable code, shouldn't it?

Reply:
it is dirty. our tests set BaseWallet injection to None and check for HTTPForbidden error. I believe the pattern was to use HTTP error for wallet access restriction and report all others in the return result. I do not believe this code will execute outside of tests, I think other things would break if BaseWallet instance was None. I did not want to change tests to make sure our API is not changed.
Suggestions? We could hide the dirty by combining Exception catching.

Reply:
As in tests that were added upstream?